dipy/workflow/segment.py

```diff
@@ -0,0 +1,39 @@
+from __future__ import division, print_function, absolute_import
+
+from glob import glob
+from os.path import join, basename, splitext
+
+import nibabel as nib
+import numpy as np
+
+from dipy.segment.mask import median_otsu
+
+
+def median_otsu_bet(input_file, out_dir, save_masked=False, median_radius=4,
+                numpass=4, autocrop=False, vol_idx=None, dilate=None):
+
+    inputs = glob(input_file)
+    out_dirs = glob(out_dir)
+    if len(out_dirs) == 1:
+        out_dirs = list(out_dirs) * len(inputs)
+
+    for (fpath, out_dir_path) in zip(inputs, out_dirs):
```

Comment:
So, here zipping the out_dirs I think it is not necessary.
You just want to go through the inputs copy their directory name and join it with the out_dirs name.